tests/RobotFramework/tests/cumulocity/configuration/configuration_operation_multi_cloud.robot

```diff
@@ -0,0 +1,115 @@
+
+
+*** Test Cases ***    DEVICE    EXTERNALID    CONFIG_TYPE    DEVICE_FILE    FILE    PERMISSION    OWNERSHIP
+#
+# Get configuration
+#
+
+Get Configuration from Main Device
+    [Template]    Get Configuration from Device
+    Text file    topic_prefix=c8y    external_id=${PARENT_SN}    config_type=CONFIG1    device_file=/etc/config1.json
+    Binary file    topic_prefix=c8y    external_id=${PARENT_SN}    config_type=CONFIG1_BINARY    device_file=/etc/binary-config1.tar.gz
+
+Get Configuration from Second Device
+    [Template]    Get Configuration from Device
+    Text file    topic_prefix=c8y-second    external_id=${SECOND_DEVICE_SN}    config_type=CONFIG1    device_file=/etc/config1.json
+    Binary file    topic_prefix=c8y-second    external_id=${SECOND_DEVICE_SN}    config_type=CONFIG1_BINARY    device_file=/etc/binary-config1.tar.gz
+
+
+*** Keywords ***
+Get Configuration from Device
+    [Arguments]    ${description}    ${topic_prefix}    ${external_id}    ${config_type}    ${device_file}
+    Log    Test Description: ${description}
+    Cumulocity.Set Device    ${external_id}
+    ${operation}=    Cumulocity.Get Configuration    ${config_type}
+    ${operation}=    Operation Should Be SUCCESSFUL    ${operation}    timeout=20
+
+    ${expected_checksum}=    Execute Command    md5sum '${device_file}' | cut -d' ' -f1    strip=${True}
+    ${events}=    Cumulocity.Device Should Have Event/s
+    ...    minimum=1
+    ...    maximum=1
+    ...    type=${config_type}
+    ...    with_attachment=${True}
+    ${contents}=    Cumulocity.Event Should Have An Attachment
+    ...    ${events[0]["id"]}
+    ...    expected_md5=${expected_checksum}
+
+    ${event}=    Cumulocity.Event Attachment Should Have File Info
+    ...    ${events[0]["id"]}
+    ...    name=^${external_id}_[\\w\\W]+${topic_prefix}-mapper-\\d+$
+
+    RETURN    ${contents}
+
+#
+# Setup
+#
+
+Suite Setup
+    # Parent
+    ${parent_sn}=    Setup    skip_bootstrap=${False}
+    Set Suite Variable    $PARENT_SN    ${parent_sn}
+
+    # Child
+    Setup Second Device
+
+Setup Second Device
+    ${child_sn}=    Catenate    SEPARATOR=_    ${parent_sn}    second
+    Set Suite Variable    $SECOND_DEVICE_SN    ${child_sn}
+
+    Execute Command
+    ...    tedge config set c8y@second.device.cert_path /etc/tedge/device-certs/tedge@second-certificate.pem
+    Execute Command    tedge config set c8y@second.device.key_path /etc/tedge/device-certs/tedge@second-key.pem
+    Execute Command    tedge config set c8y@second.proxy.bind.port 8002
+    Execute Command    tedge config set c8y@second.bridge.topic_prefix c8y-second
+    Execute Command    tedge config set c8y@second.url "$(tedge config get c8y.url)"
+
+    Execute Command    tedge cert create c8y@second --device-id ${child_sn}
+    Execute Command
+    ...    cmd=sudo env C8Y_USER='${C8Y_CONFIG.username}' C8Y_PASSWORD='${C8Y_CONFIG.password}' tedge cert upload c8y@second
+
+    Execute Command    tedge connect c8y@second
```

Comment:
One unforseen side effect of these "secondary device twins" are that they're never auto-deleted from the connected tenant.